### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by ntnshpig**

### CODE REVIEW

File: src/components/Auth/SignInModal.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+} from "./AuthStyles";
+
+const initialState = {
+  email: "",
+  password: "",
+  phone: "",
+  confirmPassword: "",
+};
+
+const phoneRegex: RegExp =
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignInModal.tsx]: Move to constants

File: src/components/Auth/SignInModal.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    resolver: yupResolver(schema),
+    mode: "onChange",
+  });
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignInModal.tsx]: useCallback

File: src/components/Auth/SignInModal.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  const [checked, setChecked] = useState(true);
+
+  const regCartIsShown = useSelector<RootState, boolean>(
+    (state) => state.ui.showReg
+  );
+  const logCartIsShown = useSelector<RootState, boolean>(
+    (state) => state.ui.showLog
+  );
+
+  const toggleHandler = () => {
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignInModal.tsx]: useMemo

File: src/components/Auth/SignInModal.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    if (logCartIsShown) dispatch(uiActions.toggleLog());
+  };
+
+  const changeSignHandler = () => {
+    dispatch(uiActions.toggleReg());
+    dispatch(uiActions.toggleLog());
+  };
+
+  const showRecoverPasHandler = () => {
+    if (regCartIsShown) dispatch(uiActions.toggleReg());
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignInModal.tsx]: Don't write inline

File: src/components/Auth/SignInModal.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+  };
+  return (
+    <>
+      <Dialog open={logCartIsShown} onClose={toggleHandler}>
+        <StyledDialogTitle>Sign In</StyledDialogTitle>
+        <Divider />
+        <Stack
+          pt={2}
+          direction="column"
+          sx={{ margin: "auto" }}
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignInModal.tsx]: Is it possible to move to JSS?

File: src/components/Auth/SignInModal.tsx [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        <Divider sx={{ marginTop: "13px" }} variant="middle">
+          OR
+        </Divider>
+        <DialogContent>
+          <form onSubmit={handleSubmit(onSubmitHandler)}>
+            <Stack direction="row">
+              <Stack direction="column" sx={{ margin: "auto" }}>
+                <StyledTypography>EMAIL*</StyledTypography>
+                <StyledTextField
+                  {...register("email")}
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignInModal.tsx]: Is it possible to avoid this?

File: src/components/Auth/SignUpModal.tsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+                  name="password"
+                  type="password"
+                  required
+                  fullWidth
+                  id="outlined-basic"
+                  variant="outlined"
+                  error={Boolean(errors.password)}
+                />
+              </Stack>
+              {regCartIsShown && (
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignUpModal.tsx]: Better use ternary expression

File: src/components/CongratModule.tsx/CongratModule.tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  const dispatch = useDispatch();
+
+  const regCartIsShown = useSelector<RootState, boolean>(
+    (state) => state.ui.congratAuth
+  );
+
+  const toggleHandler = (data: object) => {
+    dispatch(uiActions.toggleAuth());
+    dispatch(uiActions.toggleCongratAuth());
+
```
Comment (VolodymyrDevico) [inline src/components/CongratModule.tsx/CongratModule.tsx]: useless transition

File: src/components/Auth/SignUpModal.tsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+              </StyledTypographyHandler>
+            </Typography>
+          </Box>
+        </DialogActions>
+      </Dialog>
+    </>
+  );
+};
+
+export default Auth;
```
Comment (VolodymyrDevico) [inline src/components/Auth/SignUpModal.tsx]: memo

File: src/components/RecoverPassModule/RecoverPassModule.tsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+import * as yup from "yup";
+import {
+  ConfirmStyledButton,
+  StyledDialogTitle,
+  StyledTextField,
+  StyledTypography,
+  StyledTypographyHandler,
+} from "../Auth/AuthStyles";
+
+const schema = yup.object().shape({
```
Comment (VolodymyrDevico) [inline src/components/RecoverPassModule/RecoverPassModule.tsx:26]: Better to move to another component

File: src/components/RecoverPassModule/RecoverPassModule.tsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+  };
+
+  const toggleHandlerCheck = () => {
+    if (Boolean(errors.email)) return;
+
+    if (recoverIsShown) dispatch(uiActions.toggleForgetPassword());
+  };
+
+  const onSubmitHandler = (data: object) => {
+    console.log({ data });
```
Comment (VolodymyrDevico) [inline src/components/RecoverPassModule/RecoverPassModule.tsx]: Remove console log

File: src/components/RecoverPassModule/RecoverPassModule.tsx [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+  const toggleHandler = () => {
+    reset();
+    sendToggle();
+    dispatch(uiActions.toggleForgetPassword());
+  };
+
+  const toggleHandlerCheck = () => {
+    if (Boolean(errors.email)) return;
+
+    if (recoverIsShown) dispatch(uiActions.toggleForgetPassword());
```
Comment (VolodymyrDevico) [inline src/components/RecoverPassModule/RecoverPassModule.tsx]: Merge this conditions

File: src/components/Auth/AuthStyles.tsx [hunk 13]
```diff
@@ -0,0 +1,73 @@
+import {
+  Button,
+  DialogTitle,
+  Stack,
+  styled,
+  TextField,
+  Typography,
+} from "@mui/material";
+
+export const StyledButton = styled(Button)({
```
Comment (VolodymyrDevico) [inline src/components/Auth/AuthStyles.tsx:12]: Maybe you should use theme?

File: src/components/Auth/AuthStyles.tsx [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+  "&:disabled": {
+    color: "#d7d2d2",
+  },
+});
+
+export const StyledStackDescription = styled(Stack)({
+  marginTop: "15px",
+  paddingLeft: "10px",
+  alignItems: "center",
+  paddingBottom: "4px",
```
Comment (VolodymyrDevico) [inline src/components/Auth/AuthStyles.tsx]: paddingBottom: 4,

**Review by VolodymyrDevico — CHANGES_REQUESTED** [review]

**Review by turchin-aa — COMMENTED** [review]
Good job!

**Pull request merged**